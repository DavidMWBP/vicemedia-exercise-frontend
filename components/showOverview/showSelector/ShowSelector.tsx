import { Show } from "components/showOverview/show/Show";
import {
	StyledImageWrapper,
	StyledShowLabel,
	StyledShowSelector,
	StyledWrapper
} from "components/showOverview/showSelector/ShowSelector.styled";
import { motion } from "framer-motion";
import { useRouter } from "next/dist/client/router";

export interface ShowSelectorProps extends Show {
	index: number;
	isActive: boolean;
}

function ShowSelector({
	title,
	id,
	index,
	isActive,
	product_image_url
}: ShowSelectorProps): JSX.Element {
	const { push, asPath } = useRouter();

	function onShowButtonClick(id: ShowSelectorProps["id"]) {
		const pathWithShowId = `/?id=${id}`;

		if (pathWithShowId !== asPath) {
			push(pathWithShowId, pathWithShowId, { shallow: true });
		}
	}

	return (
		<StyledWrapper>
			<StyledShowLabel isActive={isActive}>{index + 1}</StyledShowLabel>
			<motion.div
				transition={{ type: "spring", bounce: 0.8, duration: 1 }}
				whileTap={{ scale: 1.2 }}
				whileHover={{ scale: 1.1 }}
			>
				<StyledImageWrapper>
					<StyledShowSelector
						src={product_image_url}
						alt={`show selector ${title}`}
						layout="fill"
						objectFit="cover"
						loading={index <= 10 ? "eager" : "lazy"}
						onClick={() => onShowButtonClick(id)}
						$isActive={isActive}
					></StyledShowSelector>
				</StyledImageWrapper>
			</motion.div>
		</StyledWrapper>
	);
}

export default ShowSelector;
