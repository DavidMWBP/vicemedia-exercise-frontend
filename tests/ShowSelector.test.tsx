import ShowSelector from "components/showOverview/showSelector/ShowSelector";
import React from "react";
import renderer from "react-test-renderer";
import showsData from "shows.json";

jest.mock("next/dist/client/router", () => ({
	__esModule: true,
	useRouter: () => ({
		push: jest.fn(() => Promise.resolve(true))
	})
}));

it("renders correctly", () => {
	const tree = renderer
		.create(<ShowSelector {...showsData.shows[5]} isActive index={5} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
