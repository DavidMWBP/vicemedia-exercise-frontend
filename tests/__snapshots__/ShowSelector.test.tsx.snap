// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
>
  <div
    className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 jBVUas"
  >
    6
  </div>
  <div
    style={Object {}}
  >
    <div
      className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
    >
      <div
        style={
          Object {
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "left": 0,
            "margin": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <img
          alt="show selector pilotweek"
          className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 gfvZOF"
          decoding="async"
          onClick={[Function]}
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fpilotweek.jpg&w=3840&q=75"
          srcSet="/_next/image?url=%2Fimages%2Fpilotweek.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
